src/getters.jl

```diff
@@ -1481,4 +1481,323 @@ function GetDrugAmounts(
+        
+    - `subject_id` - list of `subject_id`'s; each ID must be of subtype `Float64`
+        
+    - `conn` - database connection using DBInterface
+        
+    # Keyword Arguments:
+        
+    - `tab` - the `SQLTable` representing the `cohort` table; default `cohort`
+
+# Returns
+
+- `df::DataFrame` - a three column `DataFrame` comprised of columns: `:cohort_definition_id` , `:subject_id` and `:cohort_end_date`
+"""
+function GetCohortSubjectEndDate(
+    cohort_ids,
+    subject_ids,
+    conn;
+    tab = cohort
+)
+
+    df = DBInterface.execute(conn, GetCohortSubjectEndDate(cohort_ids, subject_ids; tab=tab)) |> DataFrame
+
+    return df
+    
+end
+
+"""
+function GetCohortSubjectEndDate(df:DataFrame, conn; tab = cohort)
+
+Given a `DataFrame` with a `:cohort_definition_id` column and `:subject_id` column, return the `DataFrame` with an associated `:cohort_end_date` corresponding to a given `cohort_definition_id` and `subject_id` in the `DataFrame`
+
+Multiple dispatch that accepts all other arguments like in `GetCohortSubjectEndDate(ids, conn; tab = cohort)`
+"""
+function GetCohortSubjectEndDate(
+    df::DataFrame, 
+    conn; 
+    tab = cohort
+)
+
+    return outerjoin(GetCohortSubjectEndDate(df[:,"cohort_definition_id"], df[:,"subject_id"], conn; tab=tab), df, on = :cohort_definition_id)
+
+end
+
+"""
+function GetCohortSubjectEndDate(cohort_ids; subject_ids; tab=cohort)
+
+Given a list of cohort IDs and subject IDs return their end date.
+
+# Arguments:
+
+- `cohort_ids` - list of `cohort_id`'s; each ID must be of subtype `Float64`
+    
+- `subject_id` - list of `subject_id`'s; each ID must be of subtype `Float64`
+    
+- `conn` - database connection using DBInterface
+    
+# Keyword Arguments:
+    
+- `tab` - the `SQLTable` representing the `cohort` table; default `cohort`
+   
+# Returns
+    
+- `df::DataFrame` - a three column `DataFrame` comprised of columns: `:cohort_definition_id` , `:subject_id` and `:cohort_end_date`
+
+"""
+function GetCohortSubjectEndDate(
+    cohort_ids,
+    subject_ids;
+    tab=cohort
+)
+
+    sql =
+        From(tab) |>
+        Where(Fun.in(Get.cohort_definition_id, cohort_ids...)) |>
+        Where(Fun.in(Get.subject_id, subject_ids...)) |>
+        Select(Get.cohort_definition_id, Get.subject_id, Get.cohort_end_date) |>
+        q -> render(q, dialect=dialect)
+
+    return String(sql)
+    
+end
+
+"""
+GetDatabaseCohorts(conn; tab=cohort)
+Given a `DataFrame` returns all unique cohort_definition_id associated with a database.
+
+#Arguments:
+
+- `conn` - database connection using DBInterface
+
+# Keyword Arguments:
+
+- `tab` - the `SQLTable` representing the Cohort table; default `cohort`
+
+# Returns
+    
+- `df::DataFrame` - a one column `DataFrame` comprised of columns: `:cohort_definition_id`
+
+
+"""
+function GetDatabaseCohorts(
+    conn; 
+    tab=cohort
+)
+    ids = DBInterface.execute(conn, String(GetDatabaseCohorts(tab=tab)))  |> DataFrame
```

Comment:
Ah you don't need to wrap the return of `GetDatabaseCohorts` in a String again as it already returns a string object. So please remove that conversion.